packages/flutter/lib/src/widgets/restoration_properties.dart

```diff
@@ -143,19 +143,11 @@ abstract class RestorableValue<T> extends RestorableProperty<T> {
   void didUpdateValue(T? oldValue);
 }
 
-// _RestorablePrimitiveValue and its subclasses do not allow null values in
-// anticipation of NNBD (non-nullability by default).
-//
-// If necessary, we can in the future define a new subclass hierarchy that
-// does allow null values for primitive types. Borrowing from lisp where
-// functions that returned a bool ended in 'p', a suggested naming scheme for
-// these new subclasses could be to add 'N' (for nullable) to the end of a
-// class name (e.g. RestorableIntN, RestorableStringN, etc.) to distinguish them
-// from their non-nullable friends.
-class _RestorablePrimitiveValue<T extends Object> extends RestorableValue<T> {
-  _RestorablePrimitiveValue(this._defaultValue)
-    : assert(_defaultValue != null),
-      assert(debugIsSerializableForRestoration(_defaultValue)),
+// _RestorablePrimitiveValueN and its subclasses allows for null values.
+// See [_RestorablePrimitiveValue] for the non-nullable version of this class.
+class _RestorablePrimitiveValueN<T extends Object?> extends RestorableValue<T> {
```

Comment:
Just for my own understanding, the only reason with it is `<T extends Object>` here instead of just `<T>` is because `toPrimitives` returns an `Object` and the generic has to be at least a subclass of `Object`?

Reply:
I think this just makes it explicit that T may be nullable.